datafusion/functions-array/src/string.rs

```diff
@@ -281,6 +283,49 @@ pub(super) fn array_to_string_inner(args: &[ArrayRef]) -> Result<ArrayRef> {
 
                 Ok(arg)
             }
+            Dictionary(_, _) => {
+                let any_dict_array = arr.as_any_dictionary_opt().ok_or_else(|| {
+                    DataFusionError::Internal(format!(
+                        "could not cast {} to AnyDictionaryArray",
+                        arr.data_type()
+                    ))
+                })?;
+
+                let mut values: Vec<String> = vec![];
+                macro_rules! array_function {
+                    ($ARRAY_TYPE:ident) => {{
+                        for x in downcast_arg!(any_dict_array.values(), $ARRAY_TYPE) {
+                            if let Some(x) = x {
+                                values.push(x.to_string());
+                            }
+                        }
+                    }};
+                }
+                call_array_function!(any_dict_array.values().data_type(), false);
+
+                let mut keys: Vec<usize> = vec![];
+                macro_rules! array_function {
+                    ($ARRAY_TYPE:ident) => {{
+                        for x in downcast_arg!(any_dict_array.keys(), $ARRAY_TYPE) {
+                            if let Some(x) = x {
+                                keys.push(x.to_string().parse::<usize>().unwrap());
```

Comment:
this works, but converts integers to strings just to parse them again -- I think we can just support string dictionaries and avoid this